src/storage/src/healthcheck.rs

```diff
@@ -114,20 +115,27 @@ impl PerWorkerHealthStatus {
         let mut hints: BTreeSet<String> = BTreeSet::new();
 
         for status in self.errors_by_worker.iter() {
-            use HealthStatusUpdate::*;
             for (ns, ns_status) in status.iter() {
-                if let Stalled { error, hint, .. } = ns_status {
-                    if Some(error) > namespaced_errors.get(ns).as_deref() {
-                        namespaced_errors.insert(*ns, error.to_string());
+                match ns_status {
+                    HealthStatusUpdate::Ceased { error } => {
+                        return OverallStatus::Ceased {
+                            error: error.clone(),
+                        };
```

Comment:
Can we do a similar thing as `Stalled` where we add all the `Ceased` errors to the a `namespaced` errors collection? then `Stalled` errors and hints can be added to the `hint` set?

Reply:
yeah definitely!